Native CAN device does not build when VFS is used

#### Description

Cannot build `candev_linux` when `vfs` is used.
Bisecting returned 93a521c50132e24558e42bc57dcf9c94051f07e3 as the culprit (introduced in #8652)

#### Steps to reproduce the issue

```
cd tests/conn_can
USEMODULE=vfs make BOARD=native
```

#### Expected results
Build OK

#### Actual results
```
/home/vincent/Devel/git/RIOT/cpu/native/can/candev_linux.c: In function ‘_init’:
/home/vincent/Devel/git/RIOT/cpu/native/can/candev_linux.c:165:29: error: ‘PF_CAN’ undeclared (first use in this function); did you mean ‘AF_MAX’?
     dev->sock = real_socket(PF_CAN, SOCK_RAW, CAN_RAW);
                             ^~~~~~
                             AF_MAX
/home/vincent/Devel/git/RIOT/cpu/native/can/candev_linux.c:165:29: note: each undeclared identifier is reported only once for each function it appears in
/home/vincent/Devel/git/RIOT/cpu/native/can/candev_linux.c:198:23: error: ‘AF_CAN’ undeclared (first use in this function); did you mean ‘PF_CAN’?
     addr.can_family = AF_CAN;
                       ^~~~~~
                       PF_CAN
```

#### Versions

```
Installed compiler toolchains 
-----------------------------
           native gcc: gcc (Ubuntu 7.2.0-8ubuntu3.2) 7.2.0
    arm-none-eabi-gcc: arm-none-eabi-gcc (GNU Tools for Arm Embedded Processors 7-2017-q4-major) 7.2.1 20170904 (release) [ARM/embedded-7-branch revision 255204]
              avr-gcc: missing
     mips-mti-elf-gcc: missing
           msp430-gcc: msp430-gcc (GCC) 4.6.3 20120301 (mspgcc LTS 20120406 unpatched)
                clang: clang version 4.0.1-6 (tags/RELEASE_401/final)
```

--- a/boards/native/Makefile
+++ b/boards/native/Makefile
@@ -3,3 +3,5 @@ MODULE = board
 DIRS = drivers
 
 include $(RIOTBASE)/Makefile.base
+
+INCLUDES = $(NATIVEINCLUDES)

--- a/boards/native/Makefile.include
+++ b/boards/native/Makefile.include
@@ -1,7 +1,9 @@
-export CPU = native
+export NATIVEINCLUDES += -DNATIVE_INCLUDES
+export NATIVEINCLUDES += -I$(RIOTBOARD)/$(BOARD)/include/
+export NATIVEINCLUDES += -I$(RIOTBASE)/core/include/
+export NATIVEINCLUDES += -I$(RIOTBASE)/drivers/include/
 
-# Configuration for core/include/kernel_types.h
-CFLAGS += -DNATIVE_INCLUDES
+export CPU = native
 
 USEMODULE += native-drivers
 
@@ -113,7 +115,7 @@ debug-valgrind-server: export VALGRIND_FLAGS ?= --vgdb=yes --vgdb-error=0 -v \
 term-cachegrind: export CACHEGRIND_FLAGS += --tool=cachegrind
 term-gprof: export TERMPROG = GMON_OUT_PREFIX=gmon.out $(ELFFILE)
 all-valgrind: export CFLAGS += -DHAVE_VALGRIND_H -g
-all-valgrind: export CFLAGS += $(shell pkg-config valgrind --cflags)
+all-valgrind: export NATIVEINCLUDES += $(shell pkg-config valgrind --cflags)
 all-debug: export CFLAGS += -g
 all-cachegrind: export CFLAGS += -g
 all-gprof: export CFLAGS += -pg
@@ -121,6 +123,8 @@ all-gprof: export LINKFLAGS += -pg
 all-asan: export CFLAGS += -fsanitize=address -fno-omit-frame-pointer -g
 all-asan: export CFLAGS += -DNATIVE_IN_CALLOC
 all-asan: export LINKFLAGS += -fsanitize=address -fno-omit-frame-pointer -g
+
+export INCLUDES += $(NATIVEINCLUDES)
 
 export CFLAGS += -DDEBUG_ASSERT_VERBOSE
 

--- a/boards/native/drivers/Makefile
+++ b/boards/native/drivers/Makefile
@@ -1,3 +1,5 @@
 MODULE = native-drivers
 
 include $(RIOTBASE)/Makefile.base
+
+INCLUDES = $(NATIVEINCLUDES)

--- a/cpu/native/Makefile
+++ b/cpu/native/Makefile
@@ -27,3 +27,5 @@ ifneq (,$(filter trace,$(USEMODULE)))
 endif
 
 include $(RIOTBASE)/Makefile.base
+
+INCLUDES = $(NATIVEINCLUDES)

--- a/cpu/native/Makefile.include
+++ b/cpu/native/Makefile.include
@@ -1,6 +1,8 @@
+export NATIVEINCLUDES += -I$(RIOTCPU)/native/include -I$(RIOTBASE)/sys/include
+
 # Local include for OSX
 ifeq ($(BUILDOSXNATIVE),1)
-    INCLUDES += -I$(RIOTCPU)/native/osx-libc-extra
+    export NATIVEINCLUDES += -I$(RIOTCPU)/native/osx-libc-extra
 endif
 
 USEMODULE += periph

--- a/cpu/native/mtd/Makefile
+++ b/cpu/native/mtd/Makefile
@@ -1,3 +1,5 @@
 MODULE := mtd_native
 
 include $(RIOTBASE)/Makefile.base
+
+INCLUDES = $(NATIVEINCLUDES)

--- a/cpu/native/netdev_tap/Makefile
+++ b/cpu/native/netdev_tap/Makefile
@@ -1,1 +1,3 @@
 include $(RIOTBASE)/Makefile.base
+
+INCLUDES = $(NATIVEINCLUDES)

--- a/cpu/native/socket_zep/Makefile
+++ b/cpu/native/socket_zep/Makefile
@@ -1,1 +1,3 @@
 include $(RIOTBASE)/Makefile.base
+
+INCLUDES = $(NATIVEINCLUDES)

--- a/sys/posix/include/sys/bytes.h
+++ b/sys/posix/include/sys/bytes.h
@@ -28,12 +28,7 @@
 extern "C" {
 #endif
 
-#ifndef __MACH__
 typedef size_t socklen_t;           /**< socket address length */
-#else
-/* Defined for OSX with a different type */
-typedef __darwin_socklen_t socklen_t;   /**< socket address length */
-#endif
 
 #ifdef __cplusplus
 }

--- a/sys/posix/include/sys/socket.h
+++ b/sys/posix/include/sys/socket.h
@@ -36,9 +36,6 @@
   sa_family_t sa_prefix##family
 
 #define __SOCKADDR_COMMON_SIZE  (sizeof (unsigned short int))
-#ifdef __MACH__
-#define AF_LINK (18)    /* Link layer interface */
-#endif
 #endif
 
 #include <stdlib.h>

